js/main.js

```diff
@@ -1,20 +1,147 @@
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/brandon-hoogenboom-SNxQGWxZQi0.jpg',
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/claire-rendall-b6kAwr1i0Iw.jpg',
+];
+
 const getRandomNumberSimple = (numberFrom, numberNext) => {
   if (numberFrom >= 0 && numberNext >= 0 && numberFrom < numberNext) {
     const numberMin = Math.ceil(numberFrom);
     const numberMax = Math.floor(numberNext);
     const randomSimple = Math.floor(Math.random() * (numberMax - numberMin + 1)) + numberMin;
     return Number(randomSimple);
   }
-  return new TypeError('Некоректно указанны данные');
+  throw new TypeError('Некоректно указанны данные');
 };
 
 const getRandomNumberFloat = (numberFrom, numberNext, numberFloat = 0) => {
   if (numberFrom >= 0 && numberNext >= 0 && numberFrom < numberNext && numberFloat >= 0) {
     const randomFloat = (Math.random() * (numberNext - numberFrom) + numberFrom).toFixed(numberFloat);
     return Number(randomFloat);
   }
-  return new TypeError('Некоректно указанны данные');
+  throw new TypeError('Некоректно указанны данные');
 };
 
-getRandomNumberSimple(2.1, 9.5);
-getRandomNumberFloat(6, 5, 3);
+const avatarNumbers = Array.from({ length: SIMILAR_AD_COUNT }, (v, i) => i + 1);
+
+const avatarRandomNumbers = avatarNumbers.sort(() => Math.random() - 0.5);
+
+const getAvatar = () => {
+  let avatarRandom = avatarRandomNumbers.pop();
+
+  if (avatarRandom < 10) {
+    avatarRandom = `0${String(avatarRandom)}`;
+    return avatarRandom;
+  }
+  return avatarRandom;
+};
+
+const getRandomElement = (element) => element[getRandomNumberSimple(0, element.length - 1)];
+
+const getLocationLat = () => getRandomNumberFloat(LOCATIONLAT.min, LOCATIONLAT.max, LOCATIONLAT.float);
+const getLocationLng = () => getRandomNumberFloat(LOCATIONLNG.min, LOCATIONLNG.max, LOCATIONLNG.float);
+
+const getRandomArray = (element) => {
+  const randomArray = element.sort(() => Math.random() - 0.5);
+  const createrandomArray = randomArray.slice(0, getRandomNumberSimple(1, element.length));
+  return createrandomArray;
+};
+
+const newAddItem = () => {
+  const lat = getLocationLat();
+  const lng = getLocationLng();
+
+  return {
+    author: {
+      avatar: `img/avatars/user${getAvatar()}.png`,
+    },
+    offer: {
+      title: getRandomElement(TITLES),
+      address: `${lat}, ${lng}`,
+      price: getRandomElement(PRICES),
+      type: getRandomElement(TYPES),
+      rooms: getRandomElement(ROOMS),
+      guests: getRandomElement(QUESTS),
+      checkin: getRandomElement(TIMES),
+      checkout: getRandomElement(TIMES),
+      features: getRandomArray(FEATURES),
+      description: getRandomElement(DESCRIPTIONS),
+      photos: getRandomArray(PHOTOS),
+    },
+    location: {
+      lat,
+      lng,
+    },
+  };
+};
+
+const similarItems = () => Array.from({length: SIMILAR_AD_COUNT}, newAddItem);
+
+similarItems();
```

Comment:
Здесь не нужно функцию создавать. Название similarItems и не подошло бы для функции
```suggestion
const similarItems = Array.from({length: SIMILAR_AD_COUNT}, newAddItem);
```